tests/components/gdacs/test_config_flow.py

```diff
@@ -0,0 +1,130 @@
+"""Define tests for the GDACS config flow."""
+from datetime import timedelta
+
+from asynctest import CoroutineMock, patch
+import pytest
+
+from homeassistant import data_entry_flow
+from homeassistant.components.gdacs import (
+    CONF_CATEGORIES,
+    DOMAIN,
+    FEED,
+    async_setup_entry,
+    async_unload_entry,
+    config_flow,
+)
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_RADIUS,
+    CONF_SCAN_INTERVAL,
+    CONF_UNIT_SYSTEM,
+)
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def config_entry():
+    """Create a mock GDACS config entry."""
+    return MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_LATITUDE: -41.2,
+            CONF_LONGITUDE: 174.7,
+            CONF_RADIUS: 25,
+            CONF_UNIT_SYSTEM: "metric",
+            CONF_SCAN_INTERVAL: 300.0,
+            CONF_CATEGORIES: [],
+        },
+        title="-41.2, 174.7",
+    )
+
+
+async def test_duplicate_error(hass, config_entry):
+    """Test that errors are shown when duplicates are added."""
+    conf = {CONF_LATITUDE: -41.2, CONF_LONGITUDE: 174.7, CONF_RADIUS: 25}
+
+    config_entry.add_to_hass(hass)
+    flow = config_flow.GdacsFlowHandler()
```

Comment:
We want to use the core interfaces to set up the tests to make them more robust. Don't create the flow directly.

See the hue config flow tests for good examples.

Reply:
OK, makes sense.